app/assemblers/combine_authors_and_contributors.rb

```diff
@@ -1,21 +1,24 @@
 class CombineAuthorsAndContributors
   include Mandate
 
-  initialize_with :authors, :contributors, limit: 3
+  initialize_with :authors, :contributors, limit: 3, user_id_column: :id
 
   def call
     users = randomize(authors, limit)
-    users += randomize(contributors, limit - users.count)
+
+    updated_limit = limit - users.count
+    users += randomize(contributors, updated_limit) if updated_limit.positive?
```

Comment:
This is an especially useful micro optimization for the build pages, as most categories will have at least 3 authors so in all those cases the additional query won't be executed.